include/hip/spirv_hip.hh

```diff
@@ -146,10 +146,15 @@ extern const __device__ __attribute__((weak)) __hip_builtin_gridDim_t gridDim;
 
 /* FIXME: Is this the best place for these declarations? */
 
+#ifdef __cplusplus
 extern "C" __device__ int printf(const char *fmt, ...)
     __attribute__((format(printf, 1, 2)));
 extern "C" __device__ void abort();
-
+#else
+extern __device__ int printf(const char *fmt, ...)
+    __attribute__((format(printf, 1, 2)));
+extern __device__ void abort();
```

Comment:
Is it necessary to define these for C applications? At least, the `__device__ `attribute is defunct in C mode.